File: simple_calculator.cpp
#include <iostream>

using namespace std;


// Available math operations
enum mathOperation {add=1,subtract,multiply,divide};

// Welcome the user to the calculator program
void welcome_message();
// Asks the user to enter a number, returns the number
int ask_for_num();
// Asks the user to enter an operation, performs the operation, returns the result
void perform_operation(double, double);

main()
{
    welcome_message();

    char calculate = 'Y';
    do 
    {
        double a = ask_for_num(), b = ask_for_num();
        perform_operation(a, b);
    
        cout << "Continue calculating? (Y? Only Y or y will continue...) " << endl;
        cin >> calculate;
    } while (calculate == 'Y' || calculate == 'y');
}

void welcome_message()
{
    cout << "##### ----------------- SIMPLE CALCULATOR ---------------- #####" << endl;
    cout << "----------------------------------------------------------------" << endl;
    cout << "Welcome! This simple calculator can perform simple calculations." << endl;
    cout << "You will be asked to enter two numbers and then an operation... " << endl;
    cout << "... Lets start!" << endl;
    cout << endl << endl;
}

int ask_for_num()
{
    double a;
    cout << "Enter a number:    ";
    cin >> a;

    return a;
}

void perform_operation(double a, double b)
{
    int choice;
    cout << "Available mathematical operations:    " << endl;
    cout << "1. + : addition" << endl;
    cout << "2. - : subtraction" << endl;
    cout << "3. * : multiplication" << endl;
    cout << "4. / : division" << endl;
    cout << "Enter the number of an operation:    ";
    cin >> choice;

    mathOperation operation = mathOperation(choice);
    
    switch(operation)
        {
            case add:
                cout << a << " + " << b << " = " << a + b << endl;
                break;
            case subtract:
                cout << a << " - " << b << " = " << a - b << endl;
                break;
            case multiply:
                cout << a << " * " << b << " = " << a * b << endl;
                break;
            case divide:
                if (b)
                    cout << a << " / " << b << " = " << a / b << endl;
                else
                    cout << "Cannot divide by 0!" << endl;
                break;
            default:
                cout << "That operation is not supported!" << endl;
        }
}

File: simple_calculator.py
import os


def welcome_message():
    print('''
##### ----------------- SIMPLE CALCULATOR ---------------- #####
----------------------------------------------------------------
Welcome! This simple calculator can perform simple calculations.
You will be asked to enter two numbers and then an operation... 
... Lets start!
    ''')


def ask_for_num() -> int:
    return int(input("Enter a number: "))


def ask_for_op() -> str:
    print('''
Available mathematical operations:    
+: addition
-: subtraction
*: multiplication
/: division
    ''')
    return input("Enter an operation: ")


if __name__ == "__main__":
    calculate: bool = True

    welcome_message()

    while calculate:
        a, b = ask_for_num(), ask_for_num()

        operation: str = ask_for_op()

        if operation == '+':
            print(f"{a} + {b} = {a + b}")
        elif operation == '-':
            print(f"{a} - {b} = {a - b}")
        elif operation == '*':
            print(f"{a} * {b} = {a * b}")
        elif operation == '/':
            if b:
                print(f"{a} / {b} = {a / b}")
            else:
                print("Cannot divide by zero!")
        else:
            print(f"The operation: '{operation}', is not supported!")

        # CHECK IF USER WANTS TO EXIT
        decision = input("Keep calculating? (Y/N) ")
        if decision == 'n' or decision == 'N':
            calculate = False
        elif decision == 'y' or decision == 'Y':
            continue
        else:
            print(f"'{decision}' is not valid input! Exiting...")
            break
